buildbot/master.cfg

```diff
@@ -578,6 +578,45 @@ def make_android_package(mode="normal"):
 
     return f
 
+def make_dolphin_freebsd_build(mode="normal"):
+    f = BuildFactory()
+
+    mode = mode.split(",")
+
+    nogui = "nogui" in mode
+
+    f.addStep(GitNoBranch(repourl="https://github.com/dolphin-emu/dolphin.git",
+                          progress=True, mode="incremental"))
+
+    f.addStep(ShellCommand(command=["mkdir", "-p", "build"],
+                           logEnviron=False,
+                           description="mkbuilddir",
+                           descriptionDone="mkbuilddir"))
+
+    cmake_cmd = ["cmake", ".."]
+    if nogui:
+        cmake_cmd.append("-DDISABLE_WX=ON")
+    f.addStep(ShellCommand(command=cmake_cmd,
+                           workdir="build/build",
+                           description="configuring",
+                           descriptionDone="configure",
+                           haltOnFailure=True,
+                           env={ "WX_CONFIG": "wxgtk2u-3.0-config" }))
+
+    f.addStep(Compile(command=["make", "-j4"],
```

Comment:
You could install [Ninja](https://martine.github.io/ninja/) and avoid CPU-specific settings.